### PR TITLE
Add inference_id to batches created by AL

### PR DESCRIPTION
**Pull request by robiscoding**
# Description

In order to display inference images in the model monitoring dashboard, we need to add the inference id to the images created from inference requests via the AL workflow

Relates to: https://github.com/roboflow/roboflow/pull/3080

## Type of change

Please delete options that are not relevant.

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [X] New feature (non-breaking change which adds functionality)
-   [ ] This change requires a documentation update

## How has this change been tested, please provide a testcase or example of how you tested the change?

- Test case for staging: inference_ids are sent as part of batch request (e2e test case will be tested once https://github.com/roboflow/roboflow/pull/3080 is merged)

## Any specific deployment considerations

For example, documentation changes, usability, usage/costs, secrets, etc.

## Docs

-   [ ] Docs updated? What were the changes:

### CODE REVIEW

File: inference/core/active_learning/core.py [hunk 1]
```diff
@@ -53,6 +53,7 @@ def execute_datapoint_registration(
     configuration: ActiveLearningConfiguration,
     api_key: str,
     batch_name: str,
+    inference_id=None,
```
Comment (PawelPeczek-Roboflow) [inline inference/core/active_learning/core.py:56]: typing hint required

**Comment by PawelPeczek-Roboflow** [comment]
Do we still need it?

**Comment by robiscoding** [comment]
> Do we still need it?

By it do you mean do we still need the inference_id? If so, then yes we do. I still need to fix the failing test before this is ready for review (currently a Draft/WIP)

**Comment by robiscoding** [comment]
Ready for review

**Comment by PawelPeczek-Roboflow** [comment]
I do not see `inference_id` being passed by the caller. Is that intentional?

**Review by PawelPeczek-Roboflow — APPROVED** [review]
Ok, I made mistake in previous comment overlooking actual usage. LGTM

**Pull request merged**